clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignor.java

```diff
@@ -162,6 +162,28 @@ private boolean allSubscriptionsEqual(Set<String> allTopics,
         return isAllSubscriptionsEqual;
     }
 
+    // visible for testing
+    MemberData memberDataFromSubscription(Subscription subscription) {
```

Comment:
> @showuon This method is getting less and less generic. Would it make sense to push to version handling into `memberData` method? This would let each assignor handle subscription v2 based on their specifics instead of having those conditions here.

I agree it makes it much clear to move the version handling into `memberData` method in each assignor. I've updated the PR. Thanks.